engine/src/main/java/io/zeebe/engine/processing/incident/CreateIncidentProcessor.java

```diff
@@ -41,51 +45,52 @@ public boolean onCommand(
       final CommandControl<IncidentRecord> commandControl) {
     final IncidentRecord incidentEvent = command.getValue();
 
-    final boolean incidentIsNotRejected = !rejectIncidentCreation(incidentEvent, commandControl);
+    final boolean incidentIsNotRejected = !tryRejectIncidentCreation(incidentEvent, commandControl);
 
     if (incidentIsNotRejected) {
-      final long incidentKey = commandControl.accept(IncidentIntent.CREATED, incidentEvent);
-      zeebeState.getIncidentState().createIncident(incidentKey, incidentEvent);
+      commandControl.accept(IncidentIntent.CREATED, incidentEvent);
```

Comment:
Nothing we need to do now but just thought we could also invert the logic, like rename method to createIncident, which can return true or false and on false it means we are not able to create an incident and reject it.

Reply:
Agreed, but we should probably also unimplement this as a CommandProcessor, might be nice cleanup work :)